item-impl/src/main/java/com/example/auction/item/impl/PItemState.java

```diff
@@ -54,4 +54,19 @@ public PItemStatus getStatus() {
         return item.map(PItem::getStatus).orElse(PItemStatus.NOT_CREATED);
     }
 
```

Comment:
I see you point but `PItemState` in particular wold be hard to get rid of. `PItemState` holds a `PItem` and a `PItemStatus` and also provides few methods to make state transitions during the auction easier. But I'll keep an eye on anaemic States that could be replaced with an `Optional<T>`.